appcenter-data/android/src/main/java/com/microsoft/appcenter/reactnative/data/AppCenterReactNativeDataModule.java

```diff
@@ -68,22 +68,45 @@ public void read(String documentId, String partition, ReadableMap readOptionsMap
         } else {
             readOptions = new ReadOptions(TimeToLive.DEFAULT);
         }
-        Data.read(documentId, JsonElement.class, partition, readOptions).thenAccept(new Consumer("Read failed", promise));
+        Data.read(documentId, JsonElement.class, partition, readOptions).thenAccept(new Consumer<JsonElement>("Read failed", promise));
     }
 
     @ReactMethod
-    public void create(String documentId, ReadableMap readableMap, String partition, ReadableMap writeOptionsMap, final Promise promise) {
+    public void create(String documentId, ReadableMap documentMap, String partition, ReadableMap writeOptionsMap, final Promise promise) {
         WriteOptions writeOptions;
         if (writeOptionsMap.hasKey(TIME_TO_LIVE_KEY)) {
             writeOptions = new WriteOptions(writeOptionsMap.getInt(TIME_TO_LIVE_KEY));
         } else {
             writeOptions = new WriteOptions(TimeToLive.DEFAULT);
         }
-        JsonObject jsonObject = AppCenterReactNativeDataUtils.convertReadableMapToJsonObject(readableMap);
-        Data.create(documentId, jsonObject, JsonElement.class, partition, writeOptions).thenAccept(new Consumer("Create failed", promise));
+        JsonObject jsonObject = AppCenterReactNativeDataUtils.convertReadableMapToJsonObject(documentMap);
+        Data.create(documentId, jsonObject, JsonElement.class, partition, writeOptions).thenAccept(new Consumer<JsonElement>("Create failed", promise));
     }
 
-    private class Consumer implements AppCenterConsumer<DocumentWrapper<JsonElement>> {
+    @ReactMethod
+    public void remove(String documentId, String partition, ReadableMap writeOptionsMap, final Promise promise) {
+        WriteOptions writeOptions;
+        if (writeOptionsMap.hasKey(TIME_TO_LIVE_KEY)) {
+            writeOptions = new WriteOptions(writeOptionsMap.getInt(TIME_TO_LIVE_KEY));
+        } else {
+            writeOptions = new WriteOptions(TimeToLive.DEFAULT);
+        }
+        Data.delete(documentId, partition, writeOptions).thenAccept(new Consumer<Void>("Delete failed", promise));
+    }
+
+    @ReactMethod
+    public void replace(String documentId, ReadableMap documentMap, String partition, ReadableMap writeOptionsMap, final Promise promise) {
```

Comment:
Let's align the order of parameters with `read` and `remove` APIs.
```suggestion
    public void replace(String documentId, String partition, ReadableMap documentMap, ReadableMap writeOptionsMap, final Promise promise) {
```

Reply:
Done